pkg/utils/utils.go

```diff
@@ -106,9 +104,21 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 		logger.WithError(err).Errorf("Failed to get k8s secret, %s", vsphere_secrets)
 		return err
 	}
+	// No kv pairs in the secret.
+	if len(secret.Data) == 0 {
```

Comment:
can we integrate the if block with the one above? like,

```
if err != nil || len(secret.Data) == 0 {
```

Reply:
But that wouldn't show the users the right error, there could be a valid secret, but no data populated in it and I'd like to convey that specific error scenario to the user.

Reply:
both case print out the same error message:
```
("Failed to get k8s secret, %s", vsphere_secrets)
```
Integrate this two `if` block and change the error msg into
```
("Failed to get k8s secret, %s. Error: %v", vsphere_secrets, err)
```
There should no behavior change.

Alternatively, you can consider to change error message for the second if block. e.g.
("Failed to get any data in k8s secret, %s", vsphere_secrets)

Reply:
sorry, copypasta problem, fixed the message.